sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -266,22 +261,27 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession with SQLHelper
       case _ =>
         // Create a test case to run this case.
         test(testCase.name) {
-          runTest(testCase)
+          runSqlTestCase(testCase, listTestCases, runQueries)
         }
     }
   }
 
-  /** Run a test case. */
-  protected def runTest(testCase: TestCase): Unit = {
-    def splitWithSemicolon(seq: Seq[String]) = {
-      seq.mkString("\n").split("(?<=[^\\\\]);")
-    }
-
-    def splitCommentsAndCodes(input: String) = input.split("\n").partition { line =>
-      val newLine = line.trim
-      newLine.startsWith("--") && !newLine.startsWith("--QUERY-DELIMITER")
-    }
+  /** A test case. */
+  protected trait TestCase {
+    val name: String
+    val inputFile: String
+    val resultFile: String
+  }
 
+  /** Run a test case. */
+  protected def runSqlTestCase(
+      testCase: TestCase,
+      listTestCases: Seq[TestCase],
+      runQueries: (
```

Comment:
why do we pass it as a parameter? We can directly invoke `runQueries` in `runSqlTestCase`

Reply:
Good point, I removed this and just call `runQueries` directly now.